proter/src/main/scala/com/workflowfm/proter/flows/Flow.scala

```diff
@@ -330,3 +333,57 @@ class FlowSimulationGenerator(baseName: String, flow: Flow) extends SimulationGe
     new FlowSimulation(name, manager, flow.copy())
   }
 }
+
+  /**
+    * Factory for making random flows given some parameters
+    *
+    * @param andProbability Probability that a flow node is an AND. Must in range [0,1]
+    * @param resources The list of available resources
+    * @param numTasks Distribution determining the number of tasks in the flow
+    * @param durationTasks Distribution determining task duration
+    * @param numResources Distribution determining number of resources
+    * @param cost Distribution determining cost of a task
+    * @param priority Distribution determining priority of a task
+    */
+case class RandomFlowFactory(
+    andProbability: Float, //balance between and and then nodes
+    resources: Seq[TaskResource],
+    numTasks: Distribution = Constant(1),
+    durationTasks: Distribution = Constant(1),
+    numResources: Distribution = Constant(1),
+    cost: Distribution = Constant(0),
+    priority: Distribution = Uniform(-2,2)
+) {
+
+  def withResources(r: Seq[TaskResource]): RandomFlowFactory = copy(resources=r)
+  def withTasks(d: Distribution): RandomFlowFactory = copy(numTasks=d)
+  def withDurations(d: Distribution): RandomFlowFactory = copy(durationTasks=d)
+  def withNumResources(d: Distribution): RandomFlowFactory = copy(numResources=d)
+  def withRandomPriority(d: Distribution): RandomFlowFactory = copy(priority=d)
+
+  def newFlow(): Flow = {
```

Comment:
It is best to avoid naming a function so that it includes the `new` keyword. Some good alternatives:

- `create()`/`createFlow()`
- `build()`/`buildFlow()`